.github/workflows/main.yml

```diff
@@ -0,0 +1,55 @@
+name: GitHub Actions
+
+on: [push]
```

Comment:
You're right! I think to avoid problems like #3096 , we want this:

```yaml
on:
  push:
    branches:
    - master
  pull_request:
    branches:
    - master
```

source: https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662/2?u=jameslamb

So `push: branches: master` will build when we merge to `master` and `pull_request: branches: master` will only build on pull requests into `master`.

## more references:

I've been reading [Events that trigger workflows](https://help.github.com/en/actions/reference/events-that-trigger-workflows).

I think we'll get a `pull_request` event when a PR is opened ([doc](https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories)):

![image](https://user-images.githubusercontent.com/7608904/83316535-f18a9880-a1eb-11ea-8bcd-ae28f81d966e.png)

Reply:
one other thing...after this is merged, a repo admin with more permissions than me will need to make these checks required on pull requests

![image](https://user-images.githubusercontent.com/7608904/83331050-2da81280-a259-11ea-8e55-299e5697d5a1.png)

That can be done by going to `Settings` --> `Branches` and then clicking `Edit`  on `master`. Screenshots below are from a personal repo of mine, just to show the interface

![image](https://user-images.githubusercontent.com/7608904/83331066-4c0e0e00-a259-11ea-8bf3-20f63141f4dd.png)

![image](https://user-images.githubusercontent.com/7608904/83331077-6647ec00-a259-11ea-9219-7d02f4bee184.png)